agent/connect/ca/provider_vault.go

```diff
@@ -864,6 +864,36 @@ func (v *VaultProvider) setNamespace(namespace string) func() {
 	}
 }
 
+// autotidyIssuers sets Vault's auto-tidy to remove expired issuers
+// Returns a boolean on success for testing (as there is no post-facto way of
+// checking if it is set). Logs warnings on failure to set and why, returning the
+// log message for test purposes as well.
+func (v *VaultProvider) autotidyIssuers(path string) (bool, string) {
+	s, err := v.client.Logical().Write(path+"/config/auto-tidy",
+		map[string]interface{}{
+			"enabled":              true,
+			"tidy_expired_issuers": true,
+		})
+	if err != nil {
+		errStr := err.Error()
+		switch {
+		case strings.Contains(errStr, "404"):
+			errStr = "vault versions < 1.12 don't support auto-tidy"
+		case strings.Contains(errStr, "400"):
+			errStr = "vault versions < 1.13 don't support tidy_expired_issuers"
+		case strings.Contains(errStr, "403"):
+			errStr = "permission denied setting auto-tidy in vault"
```

Comment:
I'm wondering if this would point out the problem more clearly (and tie in well with the path output in the structured log message)

```suggestion
			errStr = "permission denied on auto-tidy path in vault"
```